discovery/moby/docker.go

```diff
@@ -251,24 +252,33 @@ func (d *DockerDiscovery) refresh(ctx context.Context) ([]*targetgroup.Group, er
 		}
 
 		if d.matchFirstNetwork && len(networks) > 1 {
+			var firstNetwork *network.EndpointSettings
+			var firstNetworkMode string
 			// Match user defined network
 			if containerNetworkMode.IsUserDefined() {
-				networkMode := string(containerNetworkMode)
-				networks = map[string]*network.EndpointSettings{networkMode: networks[networkMode]}
-			} else {
-				// Get first network if container network mode has "none" value.
-				// This case appears under certain condition:
-				// 1. Container created with network set to "--net=none".
-				// 2. Disconnect network "none".
-				// 3. Reconnect network with user defined networks.
-				var first string
+				firstNetworkMode = string(containerNetworkMode)
+				firstNetwork = networks[firstNetworkMode]
```

Comment:
I'm wondering if this step is still necessary, since the next network ordering is in ascending order, and then getting the first network, which I think is more consistent.